docs/features/role-based-access-control/high-level-design.md

```diff
@@ -0,0 +1,100 @@
+for roles allows for easily comparing if a given user has certain privileges (e.g. checking if a user has the role 
+`Contributor` can be easily done: `role >= Contributor (500)`).
+
+There is also an implicit relationship between platform actions and the supported roles. By default, it is assumed that:
+  * a user with role `Viewer` can read the information for the given resource (which usually includes calls to 
+`GET resource` and `GET resource/:id` endpoints).
+  * user with role `Contributor` can perform all the actions a `Viewer` can, as well as edition actions for the given
+resource (which usually includes calls to `PATCH/PUT resource/:id` endpoints).
+  * user with `Owner` role can perform all the actions a `Contributor` can, as well as deleting actions for the given
+resource (which usually includes calls to `DELETE resource/:id` endpoints).
+
+**However, it is up to each controller method to correctly check permissions for the action being performed.** This also applies
+to custom actions that might be subject to RBAC: for instance, if publishing a project is a custom action that should be
+subject to RBAC, then it is up to the controller action to check that the user performing the request has the required role 
+(e.g. `Owner`) to perform that action.
+
+The mapping between the different platform actions and the role required to perform such action is kept in the following
+spreadsheet: https://docs.google.com/spreadsheets/d/1vz-YHdoAAG2nv_B1QUR7LQ-eU_DnKBexrqULrzMMOtw/edit?usp=sharing
+
+## Association between roles and users for resources
+
+The association between roles and users for resources will be stored in database tables in the `marxan-api` database,
+using one database table for each resource:
+* `users_organizations` will store the roles of users inside organizations.
+* `users_projects` will store the roles of users inside projects.
+* `users_scenarios` will store the roles of users inside scenarios.
+
+These tables should contain columns for `user_id`, `role` and the resource id (`organization_id`, `project_id` and 
+`scenario_id`, respectively).
+
+## Implementation details
+
+In terms of implementation, the existing module `projects-acl` should be renamed to a more generic name such as 
+`access-control` or `rbac` (though I'm not a fan of acronyms as names of modules). This module will be used across 
+the platform as the "single-source-of-truth" to check whether users are allowed to access given resources.
+
+This module, for now, will support checking permissions for the 3 resources subject to RBAC: organizations, projects and
+scenarios. For this, this module should provide 3 services, each of them implementing the following interface:
+
+
+`isOwner(resourceId: string, userId: string): boolean`
+
+Checks if the given user can perform the actions of a `Owner` for the provided resource id. This method should return 
+`true` if one of the following conditions apply:
```

Comment:
I think it would be a good idea to add abstract classes/interfaces within this PR to have more "verbose" visuals instead describing them. What do you think?

Reply:
I am describing an interface here - what is your suggestion?

Reply:
@ikas put the actual code in the repository ;)

Reply:
Oh, I see... I assumed we wanted to refrain from all-things coding related before agreeing on the implementation, but in this case I think that's a good idea to better understand the interface, I'll do that, thanks :+1:

Reply:
@ikas it could be a part of Low-level-design doc but as we already touched it, why not? also, abstract classes/interfaces aren't any real implementation ;)